workers/ohsome_quality_analyst/api/request_models.py

```diff
@@ -0,0 +1,116 @@
+"""
+Data models of the API request body.
+
+This module uses the library `pydantic` for data validation and
+settings management using Python type hinting.
+
+Besides data validation through `pydantic`, `FastAPI` will display additional
+information derived from `pydantic` models in the automatic generated API documentation.
+"""
+
+from enum import Enum
+from json import JSONDecodeError
+from typing import Optional
+
+import geojson
+import pydantic
+
+from ohsome_quality_analyst.utils.definitions import (
+    get_dataset_names,
+    get_fid_fields,
+    get_indicator_names,
+    get_layer_names,
+    get_report_names,
+)
+from ohsome_quality_analyst.utils.helper import snake_to_lower_camel
+
+IndicatorEnum = Enum("IndicatorEnum", {name: name for name in get_indicator_names()})
+ReportEnum = Enum("ReportEnum", {name: name for name in get_report_names()})
+LayerEnum = Enum("LayerNames", {name: name for name in get_layer_names()})
+DatasetEnum = Enum("DatasetNames", {name: name for name in get_dataset_names()})
+FidFieldEnum = Enum("FidFieldEnum", {name: name for name in get_fid_fields()})
+
+
+class BaseRequestModel(pydantic.BaseModel):
+    bpolys: Optional[str]
+    dataset: Optional[DatasetEnum]
+    feature_id: Optional[str]
+    fid_field: Optional[FidFieldEnum]
+
+    @pydantic.root_validator(pre=True)
+    @classmethod
+    def check_bpolys_or_dataset_and_feature_id(cls, values):
+        """Make sure either bpolys or dataset and feature_id are given as parameters."""
+        # featureId is an pydantic alias for feature_id (See config class below)
+        if (
+            "bpolys" in values
+            and "dataset" not in values
+            and not any(v in values for v in ("featureId", "feature_id"))
```

Comment:
Why not one of those:
```suggestion
            "bpolys" in values
            and not any(v in values for v in ("dataset", "featureId", "feature_id"))
```
```suggestion
            "bpolys" in values
            and not set(values).intersection(("dataset", "featureId", "feature_id"))
```

Reply:
Thanks for the throughout explanation! What I got now is following code:

```python
        _values = list(map(snake_to_lower_camel, values))
        if (
            "bpolys" in _values
            and not any(v in _values for v in ("dataset", "featureId"))
        ) or (
            "bpolys" not in _values
            and all(v in _values for v in ("dataset", "featureId"))
        ):
            return values
```